### PR TITLE
[vslib]: adapt macsec sai 1.7.1

### PR DESCRIPTION
**Pull request by Pterosaur**
Partial attributes were moved from MACsec SA to MACsec SC, Read these attributes from MACsec SC.

Signed-off-by: Ze Gan <ganze718@gmail.com>

### CODE REVIEW

**Comment by Pterosaur** [comment]
retest this please

**Review by lguohan — APPROVED** [review]

**Comment by lguohan** [comment]
retest this please

File: vslib/src/SwitchStateBaseMACsec.cpp [hunk 1]
```diff
@@ -488,16 +493,20 @@ sai_status_t SwitchStateBase::loadMACsecAttrFromMACsecSA(
...
+    attrs[2].id = SAI_MACSEC_SC_ATTR_ENCRYPTION_ENABLE;
+    attrs[3].id = SAI_MACSEC_SC_ATTR_MACSEC_CIPHER_SUITE;
 
     CHECK_STATUS(get(SAI_OBJECT_TYPE_MACSEC_SC, attr->value.oid, static_cast<uint32_t>(attrs.size()), attrs.data()));
 
     auto flow_id = attrs[0].value.oid;
     auto sci = attrs[1].value.u64;
     std::stringstream sciHexStr;
+    macsecAttr.m_encryptionEnable = attrs[2].value.booldata;
+    bool is_sak_128_bit = (attrs[3].value.s32 == SAI_MACSEC_CIPHER_SUITE_GCM_AES_128 || attrs[3].value.s32 == SAI_MACSEC_CIPHER_SUITE_GCM_AES_XPN_128) ? true : false;
```
Comment (kcudnik) [inline vslib/src/SwitchStateBaseMACsec.cpp:509]: bool is_sak_128_bit = (attrs[3].value.s32 == SAI_MACSEC_CIPHER_SUITE_GCM_AES_128 || attrs[3].value.s32 == SAI_MACSEC_CIPHER_SUITE_GCM_AES_XPN_128) 
no need for explicit true/false at the end

**Comment by kcudnik** [comment]
It would be helpful if you would add unittests recordings for this, and configuration to help prevent such issues in the future. How are you testing this right now ?

**Review by kcudnik — APPROVED** [review]
please fix build if related

File: vslib/src/SwitchStateBaseMACsec.cpp (hunk 1 above)
Comment (Pterosaur) [reply vslib/src/SwitchStateBaseMACsec.cpp]: Good suggestion, Fixed it.

**Comment by Pterosaur** [comment]
> please fix build if related

It looks like that other components are causing this failure.

**Comment by Pterosaur** [comment]
> It would be helpful if you would add unittests recordings for this, and configuration to help prevent such issues in the future. How are you testing this right now ?

Yes, There should be some unitests to test this feature.
Right now, I have a test script to test macsec orchagent. That will call sairedis indirectly. So I guess I can write a script for sairedis by leveraging the interactions between orchagent and sairedis.

**Comment by kcudnik** [comment]
sure, you can add PR with UT next

**Comment by kcudnik** [comment]
retest vs please

**Comment by Pterosaur** [comment]
retest vs please

**Comment by lguohan** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

**Review by lguohan — APPROVED** [review]

**Review by kcudnik — APPROVED** [review]

**Pull request merged**